File: generated-code/ruby/codecraft/FileReadWrite/model/debug_interface/primitive_type.rb
module Model
module DebugInterface

# Primitive type for debug rendering
module PrimitiveType
    # Lines, number of vertices should be divisible by 2
    LINES = 0
    # Triangles, number of vertices should be divisible by 3
    TRIANGLES = 1

    # Read PrimitiveType from input stream
    def self.read_from(stream)
        result = stream.read_int()
        if result < 0 || result >= 2
            raise "Unexpected tag value"
        end
        result
    end

    def self.to_s(value)
        if value == LINES
            return "LINES"
        end
        if value == TRIANGLES
            return "TRIANGLES"
        end
        raise "Impossible happened"
    end

    def self.to_str(value)
        self.to_s(value)
    end
end

end
end

File: generated-code/go/codecraft/FileReadWrite/model/debug_interface/primitive_type.go
package debug_interface

import "io"
import . "trans_gen_test/stream"

// Primitive type for debug rendering
type PrimitiveType int32

const (
    // Lines, number of vertices should be divisible by 2
    PrimitiveTypeLines PrimitiveType = 0
    // Triangles, number of vertices should be divisible by 3
    PrimitiveTypeTriangles PrimitiveType = 1
)

// Read PrimitiveType from reader
func ReadPrimitiveType(reader io.Reader) PrimitiveType {
    switch ReadInt32(reader) {
    case 0:
        return PrimitiveTypeLines
    case 1:
        return PrimitiveTypeTriangles
    }
    panic("Unexpected tag value")
}

// Get string representation of PrimitiveType
func PrimitiveTypeToString(primitiveType PrimitiveType) string {
    switch primitiveType {
    case PrimitiveTypeLines:
        return "Lines"
    case PrimitiveTypeTriangles:
        return "Triangles"
    }
    panic("Impossible happened")
}